src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -693,7 +679,7 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
             _builder.EmitIntConstant(elementCount);
             _builder.EmitOpCode(ILOpCode.Newarr);
             EmitSymbolToken(arrayType.ElementType, wrappedExpression.Syntax);
-            _builder.EmitArrayBlockInitializer(data, alignment, wrappedExpression.Syntax, _diagnostics.DiagnosticBag);
+            _builder.EmitArrayBlockInitializer(data, wrappedExpression.Syntax, _diagnostics.DiagnosticBag);
```

Comment:
>_builder.EmitArrayBlockInitializer(data, wrappedExpression.Syntax, _diagnostics.DiagnosticBag); [](http://example.com/codeflow?start=12&length=95)

It looks like we could share the same blob across arrays of different element type. Do we have a test demonstrating the fact.

Reply:
Yes. The MultipleNonSize1Types_EqualDataBlobs_NoCreateSpan_UsesCachedArray_EveryTypeGetsUniqueField test validates that 9 different types all with the same data each get their own unique T[] field but all also share the same data blob.